utils/lists.py

```diff
@@ -7,33 +7,33 @@
     "You'll be the judge", 'no... (╯°□°）╯︵ ┻━┻', 'no... baka',
     'senpai, pls no ;-;'
 ]
-i = 0
-answers = []
-questions = []
-
-## @story{9} The bot reads in a list of questions to choose from.
-with open ("trivia_questions_answers.txt") as fp:
-    count = 0
-    a_count = 0
-    q_count = 0
-    
-    for line in fp:
-        count +=1
-        if count % 2 == 0: # answer
-            answers.append(line)
-            a_count+=1
-        else: # question
-            questions.append(line)
-            q_count+=1
-
-
-_all_questions = []
-for i in range(q_count):
-     _all_questions.append(QuestionAnswer(questions[i], answers[i]))
+
+
+## @story{9} The bot needs a list of questions to choose from.
+# @param filename name of file to load the questions and answers from
+def load_trivia_questions(filename: str):
+    answers = []
+    questions = []
+
+    with open(filename) as fp:
+        count = 0
+
+        for line in fp:
+            count += 1
+            if count % 2 == 0:  # answer
+                answers.append(line.rstrip())
+            else:  # question
+                questions.append(line.rstrip())
+
+    all_questions = []
+    for i in range(len(questions)):
+        all_questions.append(QuestionAnswer(questions[i], answers[i]))
+    return all_questions
```

Comment:
Just out of curiosity, was there no way to access this question list in the way that you needed to without rewriting the code inside a function? The code looks good though. Removing the a_count and q_count is more efficient for sure.

Reply:
I did this for two reasons: First and most importantly, wrapping it in a function is needed to properly unit-test the code, as otherwise there's no way for a pytest test to execute just that "unit" of code. Second, wrapping it in a function enables better code reuse, as for example a future improvement could be to have multiple "categories" of trivia in separate files that the bot can load on demand. This way the code is independent of the file name and can be called with any filename.